time/src/main/java/net/jqwik/time/internal/properties/configurators/YearMonthRangeConfigurator.java

```diff
@@ -16,12 +16,8 @@ protected boolean acceptTargetType(TypeUsage targetType) {
 	}
 
 	public Arbitrary<?> configure(Arbitrary<?> arbitrary, YearMonthRange range) {
-		if (arbitrary instanceof YearMonthArbitrary) {
-			YearMonthArbitrary yearMonthArbitrary = (YearMonthArbitrary) arbitrary;
-			return yearMonthArbitrary.between(isoDateToYearMonth(range.min()), isoDateToYearMonth(range.max()));
-		} else {
-			return arbitrary;
-		}
+		YearMonthArbitrary yearMonthArbitrary = (YearMonthArbitrary) arbitrary;
```

Comment:
if you really want to restrict it to YearMonthArbitrary instances the `arbitrary` parameter can be typed accordingly. This will prevent casting error on wrong usage.